cdflow_commands/ecs_monitor.py

```diff
@@ -0,0 +1,157 @@
+from hashlib import sha1
+from functools import lru_cache
+from time import sleep, time
+
+from cdflow_commands.logger import logger
+
+
+def build_service_name(environment, component):
+    service_name = '{}-{}'.format(environment, component)
+    if len(service_name) > 32:
+        service_name_hash = sha1(service_name.encode('utf-8')).hexdigest()
+        service_name = '{}tf{}'.format(
+            service_name[:24], service_name_hash[:4]
+        )
+    return service_name
+
+
+class ECSMonitor():
+
+    _TIMEOUT = 600
+    _ITERATION_DELAY = 15
+
+    def __init__(self, ecs_event_iterator):
+        self._ecs_event_iterator = ecs_event_iterator
+
+    def wait(self):
+        start = time()
+        for event in self._ecs_event_iterator:
+            if time() - start > self._TIMEOUT:
+                logger.error(
+                    'Deployment timed out - '
+                    'didn\'t complete within {} seconds'.format(self._TIMEOUT)
+                )
+                raise TimeoutError
+            if event.done:
+                logger.info('Deployment complete - running: {}'.format(
+                    event.running
+                ))
+                return True
+            logger.info(
+                'Deploying ECS tasks - '
+                'desired: {} pending: {} running: {} previous: {}'.format(
+                    event.desired, event.pending,
+                    event.running, event.previous_running
+                )
+            )
+
+            sleep(self._ITERATION_DELAY)
+
+
+class ECSEventIterator():
+
+    def __init__(self, cluster, environment, component, version, boto_session):
+        self._cluster = cluster
+        self._environment = environment
+        self._component = component
+        self._version = version
+        self._boto_session = boto_session
+        self._done = False
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._done:
+            raise StopIteration
+
+        deployments = self._get_deployments()
+        primary_deployment = self._get_primary_deployment(deployments)
+        release_image = self._get_release_image(
+            primary_deployment['taskDefinition']
+        )
+
+        if release_image != '{}:{}'.format(self._component, self._version):
+            raise ImageDoesNotMatchError
```

Comment:
Seems like the reason why this would happen would be if there was another deployment started overlapping with this one. Perhaps it would be good to suggest this to the user and not output a stacktrace, so they don't think it's a bug with our code?